adr/cli.py

```diff
@@ -15,129 +19,161 @@
                         help="Format to print data in, defaults to 'table'.")
     parser.add_argument('-v', '--verbose', action='store_true', default=False,
                         help="Print the query and other debugging information.")
-    parser.add_argument('--url', default='http://activedata.allizom.org/query',
+    parser.add_argument('-u', '--url', default='http://activedata.allizom.org/query',
                         help="Endpoint URL")
-    args, remainder = parser.parse_known_args(args)
-    set_active_data_url(args.url)
-    if args.verbose:
-        log.setLevel(logging.DEBUG)
-    else:
-        log.setLevel(logging.INFO)
+    # positional arguments
+    parser.add_argument('task', nargs='*', help='Task(s) to run.')
+    return parser
 
-    all_recipes = [os.path.splitext(p)[0] for p in os.listdir(RECIPE_DIR)
-                   if p.endswith('.py') if p != '__init__.py']
 
-    if args.list:
-        log.info('\n'.join(sorted(all_recipes)))
-        return
+def _check_tasks_exist(task):
+    """Checks whether the tasks argument is populated.
 
-    for recipe in args.recipes:
-        if recipe not in all_recipes:
-            log.error("recipe '{}' not found!".format(recipe))
-            continue
-        print(run_recipe(recipe, remainder, fmt=args.fmt))
+    If no tasks are found, the program is terminated.
+
+    :param list task: list of tasks to be run.
+    """
+    if len(task) == 0:
+        sys.stdout.write('Please provide at least one recipe or query to run.\n')
+        sys.exit()
+
+
+def _list(items):
+    """Runs the -l/--list command line action.
+
+    :param list items: list of items that are to be printed to the console.
+    :returns None
+    """
+    log.info('\n'.join(sorted(items)))
+    return
+
+
+def _set_logging_verbosity(is_verbose):
+    """Runs the -v/--verbose command line action.
+
+    Default logging level is INFO.
+
+    If the -v flag is provided to adr, the logging level is set to DEBUG.
+
+    :param bool is_verbose: Speciies if -v flag has been supplied when adr was invoked.
+    """
+    log.setLevel(logging.DEBUG) if is_verbose else log.setLevel(logging.INFO)
+
+
+def main(args=sys.argv[1:]):
+    """Entry point for the adr module.
+
+    When the adr module is called, this method is run.
+
+    The argument list is parsed, and the appropriate parser or subparser is created.
+
+    Using the argument list, arguments are parsed and grouped into a Namespace object
+    representing known arguments, and a remainder list representing unknown arguments.
+
+    The method then calls the appropriate method for the action specified.
+
+    Supported use cases:
+
+    $ adr recipe <recipe_name>
+    $ adr query <query_name>
+    $ adr <recipe_name>
+
+    :param list args: command-line arguments.
+    """
+    # create parsers and subparsers.
+    parser = ArgumentParser(description='Runs adr recipes and/or queries.')
+
+    if 'query' in args or 'recipe' in args:
```

Comment:
This feels a little backwards to me, how the parser is constructed shouldn't depend on the `args`. That said, I guess it works so I'm happy to land it for now :).

Reply:
I ran into challenges in trying to implement the default case where `adr <recipe_name>` runs recipes. This was my workaround, though I do not like how it checks the contents of `args` a total of 4 times in this flow.

I'll think of something better, but would prefer to land it as is for the time being.